src/NUnitFramework/framework/Attributes/PlatformAttribute.cs

```diff
@@ -56,12 +56,20 @@ public class PlatformAttribute : IncludeExcludeAttribute, IApplyToTest
         /// <param name="test">The test to modify</param>
         public void ApplyToTest(Test test)
         {
-            if (test.RunState != RunState.NotRunnable && 
-                test.RunState != RunState.Ignored && 
-                !platformHelper.IsPlatformSupported(this))
+            try
             {
-                test.RunState = RunState.Skipped;
-                test.Properties.Add(PropertyNames.SkipReason, platformHelper.Reason);
+                if (test.RunState != RunState.NotRunnable &&
+                    test.RunState != RunState.Ignored &&
+                    !platformHelper.IsPlatformSupported(this))
+                {
+                    test.RunState = RunState.Skipped;
+                    test.Properties.Add(PropertyNames.SkipReason, platformHelper.Reason);
+                }
+            }
+            catch (Exception ex)
+            {
+                test.RunState = RunState.NotRunnable;
+                test.Properties.Add(PropertyNames.SkipReason, ex.Message);
             }
```

Comment:
Weird, @CharliePoole I saw your response in email, but it isn't here. I agree with the idea that some exceptions might need more info such as stack traces and that a good solution would be a new exception. How about adding an `InvalidPlatformException`?